lnwallet/channel.go

```diff
@@ -209,6 +209,35 @@ type PaymentDescriptor struct {
 	// Settle.
 	ParentIndex uint64
 
+	// SourceRef points to an Add update in a forwarding package owned by
+	// this channel.
+	//
+	// NOTE: This field will only be populated if EntryType is Fail or
+	// Settle.
+	SourceRef *channeldb.AddRef
```

Comment:
Since we already have `ParentIndex`, is this really necessary?

Reply:
The AddRef has (commit height, pkg index) which has no correlation HTLC indexes